app/domain/authentication/authn_k8s/inject_client_cert.rb

```diff
@@ -68,10 +69,14 @@ def validate
         @validate_pod_request.(pod_request: pod_request)
       end
 
+      def set_cert_installation_path
```

Comment:
I'll let the Ruby experts give their recommendations on this one, but my sense is that the method name shouldn't be `set_X` as that's not standard for Ruby

looking at our [style guide](https://github.com/cyberark/community/blob/master/Conjur/conventions/ruby-style-guide.md), there is indeed a link to this helpful snippet: https://github.com/rubocop-hq/ruby-style-guide#accessor_mutator_method_names

Reply:
yes, i would change this method to:
```
def set_cert_installation_path
  @cert_installation_path ||= @cert_installation_path.empty? ? DEFAULT_CERT_PATH : @cert_installation_path
end
```

in the implementation above you don't need to explicitly check if `@cert_installation_path` is `nil` as the memoization will do it for you. please add UTs for this logic if you haven't.

Reply:
btw, when will `request.headers["Certificate-Install-Path"]` be empty? i also don't see any UTs for this use-case?

Reply:
I was copying the same logic as https://github.com/cyberark/conjur/blob/f61ebd54b946d4210cb537049fcfcaab90a40d93/app/domain/authentication/authn_k8s/inject_client_cert.rb#L56 since they are both headers that are doing the same thing. I could see a use case as someone adding a `Certificate-Install-Path` header to the request with no value

Reply:
I'm still having trouble creating a UT for this. In the rspec tests we haven't been able to find a good way to do this. Do you have any suggestions?

Reply:
what if you path `cert_install_path: ""`?